pic32/cores/pic32/WInterrupts.c

```diff
@@ -263,12 +265,22 @@ void __attribute__((nomips16,at_vector(_EXTERNAL_0_VECTOR),interrupt(IPL4SRS)))
 void __attribute__((interrupt(),nomips16)) ExtInt0Handler(void)
 #endif
 {
-
+	VEC_ENT_CODE(EXT_INT0,INT0IF,PIN_INT0);
```

Comment:
To put it another way, by adjusting the "long hand" version of your code to add an extra check:

    int c = intmode[EXT_INT0];

    IFS0bits.INT0IF = 0;
    if (c < CHANGE) {
        if ((intFunc[EXT_INT0] != 0) && (c == digitalRead(PIN_INT0))) {
            *intFunc[EXT_INT0];
        }
        if (c == digitalRead(PIN_INT0)) {
            IFS0bits.INT0IF = 1;
        }
    } else {
        if (intFunc[EXT_INT0] != 0) {
            *intFunc[EXT_INT0];
        }
    }